<launch>
    <arg name="camera_namespace" default="camera_center"/>

    <node name="laplacian_line_detector" pkg="rr_iarrc" type="laplacian_line_detector"
          output="screen" ns="$(arg camera_namespace)">
        <!--Minimum Area to Keep-->
        <param name="min_blob_area" type="int" value="100" />
        <!--Range of the Laplacian Area to Floodfill-->
        <param name="laplacian_threshold_min" type="int" value="-280" />
        <param name="laplacian_threshold_max" type="int" value="-10" />
        <!--Strength of Adaptive Thresholding-->
        <param name="ignore_adaptive" type="bool" value="False" />
        <param name="adaptive_mean_threshold" type="int" value="0" />

        <!--The larger the height the lower it will be on the screen (Image Height = 1280)-->
        <!--Camera_center 570, 800, 760-->
        <param name="blockSky_height" type="int" value="560" />
        <param name="blockWheels_height" type="int" value="750" />
        <param name="blockBumper_height" type="int" value="720" />

        <param name="ignore_color_low_H" type="int" value="130" />
        <param name="ignore_color_high_H" type="int" value="190" />
        <param name="ignore_color_low_S" type="int" value="100" />
        <param name="ignore_color_high_S" type="int" value="255" />
        <param name="ignore_color_low_V" type="int" value="70" />
        <param name="ignore_color_high_V" type="int" value="255" />

        <!-- <param name="ignore_color_low_H" type="int" value="-1" />
        <param name="ignore_color_high_H" type="int" value="-1" />
        <param name="ignore_color_low_S" type="int" value="-1" />
        <param name="ignore_color_high_S" type="int" value="-1" />
        <param name="ignore_color_low_V" type="int" value="-1" />
        <param name="ignore_color_high_V" type="int" value="-1" /> -->

        <param name="subscription_node" type="string" value="image_color_rect"/>
        <param name="publish_detection_node" type="string" value="lines/detection_img"/>
        <param name="publish_debug_node" type="string" value="lines/debug_img"/>
    </node>
</launch>
